net/rpmsg/rpmsg_sockif.c

```diff
@@ -402,7 +402,7 @@ static inline void rpmsg_socket_destroy_ept(
 
   if (conn->ept.rdev)
     {
-      if (conn->backlog)
+      if (conn->backlog > 0)
```

Comment:
yeah. that is why I comment mostly at each PR I review that comparison condition should be specified and `if (a)` should be used only when `a` is `bool`. I had to investigate many bugs like this that could be avoided by using an explicit comparison condition. Anyway LGTM!